cmd/influxd/restore/restore.go

```diff
@@ -212,11 +295,134 @@ func (cmd *Command) unpackMeta() error {
+			var pb EnterpriseData
+			if err := proto.Unmarshal(fileBytes, &pb); err != nil {
+				return err
+			}
+
+			metaBytes = pb.GetData()
+		} else {
+			return fmt.Errorf("Unrecognized backup platform: %s", cmd.manifest.Platform)
+		}
+
+	} else {
+		// find the meta file
+		metaFiles, err := filepath.Glob(filepath.Join(cmd.backupFilesPath, backup_util.Metafile+".*"))
+		if err != nil {
+			return err
+		}
+
+		if len(metaFiles) == 0 {
+			return fmt.Errorf("no metastore backups in %s", cmd.backupFilesPath)
+		}
+
+		fileName := metaFiles[len(metaFiles)-1]
+		cmd.StdoutLogger.Printf("Using metastore snapshot: %v\n", fileName)
+		metaBytes, err = backup_util.GetMetaBytes(fileName)
+	}
+
+	// Read the metastore backup
+	req := &snapshotter.Request{
+		Type:                   snapshotter.RequestMetaStoreUpdate,
+		BackupDatabase:         cmd.sourceDatabase,
+		RestoreDatabase:        cmd.destinationDatabase,
+		BackupRetentionPolicy:  cmd.backupRetention,
+		RestoreRetentionPolicy: cmd.restoreRetention,
+	}
+
+	resp, err := cmd.updateMetaRemote(req, bytes.NewReader(metaBytes), int64(len(metaBytes)))
+	if err != nil {
+		return err
+	}
+
+	header := binary.BigEndian.Uint64(resp[:8])
+	npairs := binary.BigEndian.Uint64(resp[8:16])
+
+	if npairs == 0 {
+		return fmt.Errorf("DB metadata not changed. database %s may already exist", cmd.destinationDatabase)
+	}
+
+	pairs := resp[16:]
+
+	if header != snapshotter.BackupMagicHeader {
+		return fmt.Errorf("Response did not contain the proper header tag.")
+	}
+
+	if (len(pairs)/8)%2 != 0 {
+		return fmt.Errorf("expected an even number of integer pairs in update meta repsonse")
+	}
+
+	cmd.shardIDMap = make(map[uint64]uint64)
+	for i := 0; i < int(npairs); i++ {
+		offset := i * 16
+		k := binary.BigEndian.Uint64(pairs[offset : offset+8])
+		v := binary.BigEndian.Uint64(pairs[offset+8 : offset+16])
+		cmd.shardIDMap[k] = v
+	}
+
+	return err
+}
+
+// takes a request object, writes a Base64 encoding to the tcp connection, and then sends the request to the snapshotter service.
+func (cmd *Command) updateMetaRemote(req *snapshotter.Request, upStream io.Reader, nbytes int64) ([]byte, error) {
+
+	req.UploadSize = nbytes
+	var err error
+
+	var b bytes.Buffer
+
+	// Connect to snapshotter service.
+	conn, err := tcp.Dial("tcp", cmd.host, snapshotter.MuxHeader)
+	if err != nil {
+		return b.Bytes(), err
+	}
+	defer conn.Close()
+
+	if _, err := conn.Write([]byte{byte(req.Type)}); err != nil {
+		return b.Bytes(), err
+	}
+
+	if req.Type != snapshotter.RequestShardUpdate { // Write the request
+		if err := json.NewEncoder(conn).Encode(req); err != nil {
+			return b.Bytes(), fmt.Errorf("encode snapshot request: %s", err)
+		}
+	}
+
+	if n, err := io.Copy(conn, upStream); (err != nil && err != io.EOF) || n != req.UploadSize {
+		return b.Bytes(), fmt.Errorf("error uploading file: err=%v, n=%d, uploadSize: %d", err, n, req.UploadSize)
+	}
+
+	cmd.StdoutLogger.Printf("wrote %d bytes", req.UploadSize)
+
+	b.Reset()
```

Comment:
`b` was declared at the beginning of this function, but received no writes, and is being `Reset` here? Why not just use `ioutil.ReadAll`? I don't see the benefit in using `Buffer` here.